server/models/Wallet.js

```diff
@@ -414,7 +414,7 @@ class Wallet{
         if(
           trustRelationship.actor_wallet_id === senderWallet.getId() &&
           trustRelationship.target_wallet_id === receiveWallet.getId() &&
-          trustRelationship.request_type === TrustRelationship.ENTITY_TRUST_REQUEST_TYPE.send
+          trustRelationship.request_type === trustType
```

Comment:
This hasTrust method in wallet model was not actually referencing the trustType parameter sent in, so I changed it to be more flexible to reuse for checking other request types